src/utils/url.ts

```diff
@@ -33,3 +37,55 @@ export const getParameterByName = (name: string, url?: string | null | undefined
     // eslint-disable-next-line compat/compat
     return new URLSearchParams(url).get(name) || '';
 };
+
+export interface ScaleImageOptions {
+    maxWidth?: number;
+    width?: number;
+    maxHeight?: number;
+    height?: number;
+    fillWidth?: number;
+    fillHeight?: number;
+    quality?: number;
+}
+
+/**
+ * Returns the url of the first or a random backdrop image of an item.
+ * If the item has no backdrop image, the url of the first or a random backdrop image of the parent item is returned.
+ * Falls back to the primary image (cover) of the item, if neither the item nor it's parent have at least one backdrop image.
+ * Returns undefined if no usable image was found.
+ * @param apiClient The ApiClient to generate the url.
+ * @param item The item for which the backdrop image is requested.
+ * @param random If set to true and the item has more than one backdrop, a random image is returned.
+ * @param options Optional; allows to scale the backdrop image.
+ * @returns The url of the first or a random backdrop image of the provided item.
+ */
+export const getItemBackdropImageUrl = (apiClient: ApiClient, item: BaseItemDto, random = false, options: ScaleImageOptions = {}): string | undefined => {
+    let imgUrl;
+
+    if (item.BackdropImageTags?.length) {
+        const backdropImgIndex = random ? randomInt(0, item.BackdropImageTags.length - 1) : 0;
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Please add a check for `item.Id` in the if statements like you did for the `ParentBackdropItemId` instead of using the non-null assertion.